src/dmd/dsymbolsem.d

```diff
@@ -2365,7 +2365,10 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             if (auto te = ed.memtype.isTypeEnum())
             {
                 EnumDeclaration sym = cast(EnumDeclaration)te.toDsymbol(sc);
-                if (!sym.memtype || !sym.members || !sym.symtab || sym._scope)
+                // Special enums like __c_[u]long[long] are fine to forward reference
+                // see https://issues.dlang.org/show_bug.cgi?id=20599
+                if (sym.isSpecial()) {}
+                else if (!sym.memtype || (!sym.members) || !sym.symtab || sym._scope)
```

Comment:
Why not just bundle it all into one condition `if (!sym.isSpecial() && (!sym.memtype || ...))`?

Reply:
done